### PR TITLE
Feat/logging system

### PR DESCRIPTION
**Pull request by bin315a1**
## Changes proposed in this pull request
1. Added winston npm module for logging
2. Changed the standard Controller error module
3. Refactored upload profile pic endpoint

@bruinpool-devs

### CODE REVIEW

File: src/noti/controller.js [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
-      const res = await Noti.find({ username }).sort({date: -1}).skip((Number(pageNum)-1)*pageLimit).limit(pageLimit);
+      const res = await Noti.find({ username })
+        .sort({ date: -1 })
+        .skip((Number(pageNum) - 1) * pageLimit)
+        .limit(pageLimit);
       return resolve(res);
-    } catch(err) {
-      return reject(Error(500));
+    } catch (err) {
+      return reject(err);
```
Comment (Erick-Suarez) [inline src/noti/controller.js:15]: Do these not use controller exception anymore?

File: src/noti/controller.js [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
-        {new: true}
+        { $set: { viewed: true, viewedAt: new Date() } },
+        { new: true }
       );
-      
+
       return resolve(res);
     } catch (err) {
-      return reject(Error(500));
+      return reject(err);
```
Comment (Erick-Suarez) [inline src/noti/controller.js:31]: And here

File: src/request/controller.js [hunk 3]
```diff
@@ -17,11 +17,13 @@ const getRequestInfo = async (requestID) => {
     try {
       const res = await Request.findById(query);
       if (res == null) {
-        reject(Error(404));
+        return reject(
+          new ControllerException(404, "ride of ride_id not found")
+        );
```
Comment (Erick-Suarez) [inline src/request/controller.js:22]: Is it better practice to be a throw inside a try catch, and the catch rejects the exception?

File: src/ride/controller.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ const createRideQueryFilter = (filter_) => {
     try {
       filter_ = JSON.parse(filter_);
     } catch (err) {
-      return reject(Error(400, err));
+      return reject(err);
```
Comment (Erick-Suarez) [inline src/ride/controller.js:22]: why no exception handling here

File: src/user/controller.js [hunk 5]
```diff
@@ -479,27 +444,27 @@ const getSchool = (username) => {
...
-  checkAvailability,
   login,
   findUserByEmail,
   findUserByUsername,
-  findUserByPhoneNumber,
-  getMyInfo,
-  uploadPicUrl,
+  // findUserByPhoneNumber,
+  // getMyInfo,
+  // uploadPicUrl,
```
Comment (Erick-Suarez) [inline src/user/controller.js:459]: why are these commented out

File: src/user/index.js [hunk 6]
```diff
@@ -1,9 +1,8 @@
 const express = require("express");
 const router = new express.Router();
 
-const multiparty = require("multiparty");
-const fileType = require("file-type");
-const fs = require("fs");
+// const fileType = require("file-type");
+// const fs = require("fs");
```
Comment (Erick-Suarez) [inline src/user/index.js:5]: whyu are these commented out instead of removed

File: src/user/index.js [hunk 7]
```diff
@@ -33,8 +30,9 @@ router.post("/users/login", async (req, res) => {
       }
     );
     res.status(200).send({ authToken: token });
-  } catch (e) {
-    res.status(401).send({ message: e });
+  } catch (err) {
+    // errResp(res, err)
```
Comment (Erick-Suarez) [inline src/user/index.js:34]: delete this

File: src/noti/controller.js (hunk 1 above)
Comment (bin315a1) [reply src/noti/controller.js:15]: Yeah the ControllerException extends its parent class Error anyway, so we can pass that in directly

File: src/request/controller.js (hunk 3 above)
Comment (bin315a1) [reply src/request/controller.js:22]: no it's much better to just return a reject at the spot without throwing

File: src/user/controller.js (hunk 5 above)
Comment (bin315a1) [reply src/user/controller.js:459]: They're either not used or deprecated

File: src/user/index.js (hunk 6 above)
Comment (bin315a1) [reply src/user/index.js:5]: b/c I forgot

**Pull request merged**